Demo/src/main/java/com/paypal/android/ui/paypal/PayPalFragment.kt

```diff
@@ -55,15 +68,134 @@ class PayPalFragment : Fragment() {
-            if (!canRunPayPalCheckout) Text(text = stringResource(id = R.string.minimum_sdk_needed))
+        val isLoading by payPalViewModel.isLoading.observeAsState(initial = false)
+        ConstraintLayout(
+            modifier = Modifier.fillMaxHeight(),
+        ) {
+            val (button, text, result) = createRefs()
+            if (isLoading) {
+                LoadingComposable(modifier = Modifier.constrainAs(result) {
+                    top.linkTo(parent.top)
+                    bottom.linkTo(button.top)
+                    start.linkTo(parent.start)
+                    end.linkTo(parent.end)
+                })
+            } else {
+                CheckoutResult(
+                    payPalViewModel.checkoutResult,
+                    modifier = Modifier.constrainAs(result) {
+                        top.linkTo(parent.top)
+                        bottom.linkTo(button.top)
+                    })
+            }
+            Button(
+                enabled = canRunPayPalCheckout && !isLoading,
+                onClick = { launchNativeCheckout() },
+                modifier = Modifier
+                    .padding(start = 16.dp, end = 16.dp, bottom = 16.dp, top = 16.dp)
+                    .fillMaxWidth()
+                    .constrainAs(button) {
+                        if (canRunPayPalCheckout) {
+                            bottom.linkTo(parent.bottom)
+                        } else {
+                            bottom.linkTo(text.top)
+                        }
+                    }
+            ) {
+                Text(stringResource(R.string.start_checkout))
+            }
+            if (!canRunPayPalCheckout) Text(
+                text = stringResource(id = R.string.minimum_sdk_needed),
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .constrainAs(text) {
+                        bottom.linkTo(parent.bottom, margin = 16.dp)
+                    },
+                textAlign = TextAlign.Center,
+            )
+        }
+    }
+
+    @Composable
+    private fun CheckoutResult(resultLiveData: LiveData<PayPalCheckoutResult>, modifier: Modifier) {
+        val result by resultLiveData.observeAsState(initial = null)
+        when (result) {
+            is PayPalCheckoutResult.Success -> CheckoutSuccess(
+                result as PayPalCheckoutResult.Success,
+                modifier = modifier
+            )
+            is PayPalCheckoutResult.Failure -> CheckoutFailure(
+                result as PayPalCheckoutResult.Failure,
+                modifier = modifier
+            )
+            PayPalCheckoutResult.Cancellation -> CheckoutCancelled(modifier = modifier)
+        }
+
+    }
+
+    @Composable
+    private fun CheckoutSuccess(result: PayPalCheckoutResult.Success, modifier: Modifier) {
+        Column(modifier = modifier.padding(horizontal = 16.dp)) {
+            Text(
+                text = "Order Approved",
+                fontSize = 24.sp,
```

Comment:
qq: Can Compose get these strings values out of the _strings.xml_? and is it worth doing it for the Demo app?

Reply:
yes it can. I'll add them

Reply:
@andrespelaezp fixed